core/tests/test_utils.py

```diff
@@ -622,6 +622,9 @@ def process_and_flush_pending_tasks(self, queue_name=None):
                 if task.url == '/_ah/queue/deferred':
                     from google.appengine.ext import deferred
                     deferred.run(task.payload)
+                elif feconf.EMAILS_TASK_PREFIX in task.url:
+                    # Flush task if task is of emails taskqueue.
```

Comment:
(Also, just to be clear, I don't mind whether we use payload or params; the important thing here is that the test has as little of its own logic as possible and accurately tests the underlying system. Otherwise we're going to run into cases where things work in dev but they break in production, and for something as behind-the-scenes as emails, it's important to guard against this happening.)